force-app/main/default/lwc/nksSurvey/nksSurvey.js

```diff
@@ -0,0 +1,111 @@
+export default class NksSurvey extends LightningElement {
+    surveyId;
+    backgroundColor;
+    title;
+    question;
+    isAnswered;
+    isRendered = false;
+    startDate;
+    endDate;
+    rating;
+    comment;
+    show = true;
+
+    @wire(getSurvey)
+    wiredSurvey({ data, error }) {
+        if (data) {
+            this.surveyId = data.Id;
+            this.backgroundColor = data.NKS_Background_Color__c;
+            this.title = data.NKS_Title__c;
+            this.question = data.NKS_Question__c;
+            this.startDate = data.NKS_Start_Date__c;
+            this.endDate = data.NKS_End_Date__c;
+        } else if (error) {
+            console.log('Problem getting survey: ' + error);
+        }
+    }
+
+    get background() {
+        return `background-color: ${this.backgroundColor}`;
+    }
+
+    get hide() {
+        return !this.show || this.isAnswered;
+    }
+
+    get isValid() {
+        const currentdate = new Date();
+        if (this.surveyId) {
+            return new Date(this.startDate) <= currentdate && new Date(this.endDate) >= currentdate ? true : false;
+        }
+        return false;
+    }
+
+    renderedCallback() {
+        if (this.surveyId && !this.isRendered) {
+            hasAnswered({ surveyId: this.surveyId }).then((res) => {
+                this.isAnswered = res;
+                this.isRendered = true;
+                console.log('Survey is answered: ', res);
+            });
+        }
+    }
+
+    handleClick(event) {
+        this.rating = event.currentTarget.getAttribute('value');
+        const currentId = 'emoji' + this.rating;
+        this.handleSelected(currentId);
+    }
+
+    handleSelected(currentId) {
+        const emojiIds = ['emoji1', 'emoji2', 'emoji3', 'emoji4', 'emoji5'];
+
+        for (const emojiId of emojiIds) {
+            const paths = this.template.querySelectorAll(`div[data-id="${emojiId}"] path`);
+
+            if (currentId === emojiId) {
+                paths[0].setAttribute('class', 'select1');
+                paths[1].setAttribute('class', 'select2');
+            } else {
+                paths[0].setAttribute('class', 'path1');
+                paths[1].setAttribute('class', 'path2');
+            }
+        }
+    }
+
+    handleChange() {
+        this.comment = this.template.querySelector('lightning-textarea').value;
+    }
+
+    handleSend() {
+        this.show = false;
+        const event = new ShowToastEvent({
+            title: 'Tilbakemeldingen din er mottatt.',
+            message: 'Ha en fin dag videre!',
+            variant: 'success'
+        });
+        this.dispatchEvent(event);
+        createAnsweredRecord({ surveyId: this.surveyId }).then((res) => {
+            console.log('Result of Survey Answered creation: ', res);
+
+            createResponseRecord({ surveyId: this.surveyId, rating: this.rating, comment: this.comment }).then(
```

Comment:
Dette trenger ikke å være to apex call, det burde bare kalle en funksjon createResponse elns. som oppreter både answered og response records. Å håndtere det i backenden e mer effektivt, kjappere og mindre ressurskrevende.

Reply:
Det kan håndteres på samme måte som i LWCen egentlig. Bare send med en boolean "isCancel" f. eks. Answered skal jo uansett opprettes, og hvis isCancel == true så opprettes ikke response.

Reply:
Det fikser nok det å kalle funksjonen en gang i lwc; det jeg ikke skjønner er hvordan dette er mer effektivt? Det å lage en stor funksjon vil øke funksjonens kompleksitet, og hvis bruker ikke svarer vil funksjonen kalles selv om vi legger til en sjekk for å ikke kjøre hele funksjonen???

Reply:
Du trenger ikke å ha bare en funksjon 😄, du bare kaller den ene funksjonen i den andre. Altså at funksjon A blir kalt i LWC og den sin logikk kaller den på funksjon B. Da er det både mer tydelig, gjenbrukbart, og effektivt. Det er mer tydelig fordi alt av logikk skjer på samme sted, når du leser apex klassen er det tydelig hva den gjør, du må ikke gå tilbake til LWCen for å forstå det. Det er mer gjenbrukbart fordi noen andre kan lage en tilfredshets frontend og bruke bare den funksjonen. Og det er mer effektivt fordi veielderen bare sender en request til backende, siden hver request må reise til tyskland.

Reply:
@saramoh2309 Det er mer effektivt ved at man ikke trenger å gå innom serveren mer enn én gang. Nå må man innom server for å kalle på første metode, vente på det, deretter gå innom enda en gang. Ved å ha ett funksjonskall tar man bare en runde til server, og that's it. Performance wise så handler det egentlig bare om latency.

Personlig synes jeg også at å ha begge tilfeller i Apex-metoden er tvertimot mindre komplekst, ettersom man slipper hele then-opplegget i lwc i bytte mot en enkel if-statement på backend. Men det er kanskje bare smak og behag.